packages/core/src/controllers/ChainController.ts

```diff
@@ -153,6 +154,17 @@ export const ChainController = {
   },
 
   removeAdapter(namespace: ChainNamespace) {
+    if (state.activeChain === namespace) {
+      const nextAdapter = Array.from(state.chains.entries()).find(
+        ([chainNamespace]) => chainNamespace !== namespace
+      )
+      if (nextAdapter) {
+        const caipNetwork = nextAdapter[1]?.caipNetworks?.[0]
```

Comment:
what is `nextAdapter[0]` ??

Reply:
Since chains is a map, `0`. index is namespace while `1`.index is adapter itself